packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import DebounceInput from 'react-debounce-input';
+import getMinutes from 'date-fns/get_minutes';
+import getHours from 'date-fns/get_hours';
+import getDate from 'date-fns/get_date';
+import getMonth from 'date-fns/get_month';
+import getYear from 'date-fns/get_year';
+import setDate from 'date-fns/set_date';
+import setMinutes from 'date-fns/set_minutes';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import isSameMinute from 'date-fns/is_same_minute';
+import startOfDay from 'date-fns/start_of_day';
+import startOfMinute from 'date-fns/start_of_minute';
+import twoDigits from '../shared/utils/format/twoDigits';
+import DateTimePicker from '../DateTimePicker';
+import theme from './InputDateTimePicker.scss';
+
+const splitPartsRegex = new RegExp(/^\s*([^\s]+?)\s+([^\s]+?)\s*$/);
+const datePartRegex = new RegExp(/^\s*([0-9]{4})-([0-9]{1,2})-([0-9]{1,2})\s*$/);
+const timePartRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const day = getDate(date);
+	const month = getMonth(date) + 1;
+	const year = getYear(date);
+
+	const dateText = `${year}-${twoDigits(month)}-${twoDigits(day)}`;
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return `${dateText} ${timeText}`;
+}
```

Comment:
`parse` is not possible because it only parse a full date, and here we can parse a date or time. There is other problems : 
```
// Seems to have no check for month boundaries
parse('2014-24-11T11:30:30') // => 2015-12-11T10:30:30.000Z
// Neither for day
parse('2014-01-42T11:30:30') // => 2014-02-11T10:30:30.000Z
```
And with l10n, there will be plenty of format, and I don't think any lib could handle all of that with a real error handling.

For format, I'll use it.